test/e2e/operator_test.go

```diff
@@ -4323,6 +4405,29 @@ func assertServiceAnnotation(t *testing.T, serviceName types.NamespacedName, ann
 	}
 }
 
+// assertLoadBalancerServiceAnnotationWithPollImmediate checks if the specified annotation on the
+// LoadBalancer Service of the given IngressController matches the expected value.
+func assertLoadBalancerServiceAnnotationWithPollImmediate(t *testing.T, kclient client.Client, ic *operatorv1.IngressController, annotationKey, expectedValue string) {
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		service := &corev1.Service{}
+		if err := kclient.Get(context.Background(), controller.LoadBalancerServiceName(ic), service); err != nil {
+			t.Logf("failed to get service %s: %v", controller.LoadBalancerServiceName(ic), err)
+			return false, nil
+		}
+		if actualValue, ok := service.Annotations[annotationKey]; !ok {
+			t.Logf("load balancer has no %q annotation: %v", annotationKey, service.Annotations)
+			return false, nil
+		} else if actualValue != expectedValue {
+			t.Logf("expected %s, found %s", expectedValue, actualValue)
+			return false, nil
```

Comment:
I expect that we don't want to keep trying, after we found an unexpected value.  Or would we expect it change after this?
```suggestion
			return false, fmt.Errorf("expected %s, found %s", expectedValue, actualValue)
```

Reply:
I think we need to keep trying here because the annotation value might not get updated immediately after the infra status is updated.